### ISSUE TITLE
wrong string conversion with xetex

### ISSUE THREAD

**Issue by u-fischer** [ho-tex/pdfescape#1]
The conversion with xetex is wrong for all chars > 256. 

~~~~
\documentclass{article}
\usepackage{pdfescape}
\begin{document}
\EdefEscapeString\blub{€}
\show\blub
\end{document}
~~~~

With pdflatex (based on the primitive `\pdfescapestring`), with lualatex (based one the lua emulation `\pdf@escapestring` in pdftexcmds) and also with platex one gets

~~~~
> \blub=macro:
->\342\202\254.
~~~~

xelatex and uplatex gives

~~~~
> \blub=macro:
->\13054.
~~~~

**Comment by davidcarlisle** [comment]
we could make the package a thin wrapper around l3?

```
\ExplSyntaxOn
\str_set_convert:Nnnn\blubb{€}{}{utf8/string}
\show\blubb
\ExplSyntaxOff
```

shows

```
> \blubb=macro:
->\342\202\254.
```

in xelatex for example (although not pdftex without a bit more work)

**Comment by u-fischer** [comment]
Yes, the main question is where to make a change. One could also adapt pdftexcmds to provide \pdf@escapestring for xetex with the help of  `\str_set_convert:Nnnn`. Then pdfescape would pick that up.
